squirrel.worker/src/main/java/org/dice_research/squirrel/analyzer/impl/html/scraper/HtmlScraper.java

```diff
@@ -219,47 +175,95 @@ public HtmlScraper() {
+        webClient.setRefreshHandler(new ThreadedRefreshHandler());
+        webClient.getCookieManager().setCookiesEnabled(true);
+        webClient.getOptions().setJavaScriptEnabled(true);
+        webClient.getOptions().setUseInsecureSSL(true);
+        webClient.getOptions().setThrowExceptionOnScriptError(false);
+        webClient.getOptions().setCssEnabled(true);
+        long timeout = getJavascriptTimeout(uri);
+        try {
+            HtmlPage htmlPage = webClient.getPage(uri.getUri().toString());
+            webClient.waitForBackgroundJavaScript(timeout);
+            this.doc = Jsoup.parse(htmlPage.getWebResponse().getContentAsString(), "UTF-8");
+        } catch (IOException e){
+            LOGGER.warn("Error in handling java script by htmlunit: " + e.getMessage());
+        }
+    }
 
+    private void handlePageLoad(CrawleableUri uri){
+        HtmlPage htmlPage;
+        WebClient webClient = new WebClient(BrowserVersion.FIREFOX_60);
+        webClient.getOptions().setJavaScriptEnabled(true);
+        webClient.getOptions().setThrowExceptionOnScriptError(true);
+        webClient.getOptions().setCssEnabled(false);
+        webClient.setAjaxController(new NicelyResynchronizingAjaxController());
+        long timeout = getJavascriptTimeout(uri);
+        try{
+            htmlPage = webClient.getPage(uri.getUri().toString());
+            DomElement btn = htmlPage.getElementById(pageLoadResources.get(YamlFileAtributes.ID).toString());
+            DomElement disabledElement = htmlPage.getElementById(pageLoadResources.get(YamlFileAtributes.ID).toString());
+            do {
+                htmlPage = btn.click();
+                webClient.waitForBackgroundJavaScript(timeout);
+            } while (!disabledElement.getStyleMap().containsKey("display") ||
+                (disabledElement.getStyleMap().containsKey("display") &&
+                    !disabledElement.getStyleMap().get("display").getValue().equals("none")));
+            this.doc = Jsoup.parse(htmlPage.getWebResponse().getContentAsString(), "UTF-8");
+        } catch (Exception e) {
+            LOGGER.error("An error occurred when trying handle page load, ", e);
+        }
+    }
 
-        List<String> resourcesList = new ArrayList<String>();
-        
+    private Set<Triple> scrapeDownloadLink(Map<String, Object> resources, File htmlFile, CrawleableUri curi) throws Exception {
+        this.doc = Jsoup.parse(htmlFile, "UTF-8");
+        this.uri = curi.getUri().toString();
+        Set<Triple> triples = new LinkedHashSet<Triple>();
+        this.label = uri.substring(uri.lastIndexOf("/")+1, uri.length());
 
-//        if(uri.contains("?")) {
-//         this.label = uri.substring(uri.lastIndexOf("/")+1, uri.lastIndexOf("?"));
-//        }else {
-//         this.label = uri.substring(uri.lastIndexOf("/")+1, uri.length());
-//        }
+        if (!htmlFile.toString().contains("test")) //To prevent downloading a page when running unit test cases
```

Comment:
Please find a better way to handle these cases. Simply checking for "test" in a URL is a little bit too error prone.